sms_sponsorship/models/sms_child_request.py

```diff
@@ -83,10 +83,14 @@ def _compute_full_url(self):
     @api.depends('date')
     def _compute_event(self):
         for request in self.filtered('date'):
-            request.event_id = self.env['crm.event.compassion'].search([
+            event_id = self.env['crm.event.compassion'].search([
                 ('accepts_sms_booking', '=', True),
-                ('start_date', '<=', request.date)
-            ], limit=1)
+                ('start_date', '<=', request.date),
+                ('start_date', '>=', datetime.today() -
```

Comment:
You should compare to a string and not to a date object, so here you should use the function `fields.Datetime.to_string(date)` to convert your datetime object.